### PR TITLE
Simplify param types for FactoryLocator.

### PR DESCRIPTION
**Pull request by ADmad**
Remove use of TableLocator class of database package from FactoryLocator.

### CODE REVIEW

File: tests/bootstrap.php [hunk 1]
```diff
@@ -108,6 +110,8 @@
     ConnectionManager::get('test')->getDriver()->enableAutoQuoting(true);
 }
 
+FactoryLocator::add('Table', new TableLocator());
```
Comment (ADmad) [inline tests/bootstrap.php]: We could add this to a new bootstrap file in the ORM package (autoloaded by composer), for friction free usage/upgrade.

Comment (markstory) [reply tests/bootstrap.php]: I think that will make our lives and our user's lives more simple. We would need the bootstrap in both the `cakephp/cakephp` and `cakephp/orm` packages.

File: src/Datasource/FactoryLocator.php [hunk 2]
```diff
@@ -28,18 +27,18 @@ class FactoryLocator
...
 
     /**
      * Register a callable to generate repositories of a given type.
      *
      * @param string $type The name of the repository type the factory function is for.
-     * @param \Cake\Datasource\Locator\LocatorInterface|callable $factory The factory function used to create instances.
+     * @param \Cake\Datasource\Locator\LocatorInterface $factory The factory function used to create instances.
      * @return void
      */
-    public static function add(string $type, LocatorInterface|callable $factory): void
```
Comment (markstory) [inline src/Datasource/FactoryLocator.php:42]: We could add a deprecation warning when we get a callable in 4.4.

File: tests/bootstrap.php (hunk 1 above)
Comment (ADmad) [reply tests/bootstrap.php]: Done

**Pull request merged**